### PR TITLE
feat(linear): Support emojis for projects

### LINKED ISSUE #2695: Linear: Missing emojis for projects
# Extension – [Linear](https://www.raycast.com/raycast/linear)

Author: @thomaslombart 

## Description

Looks like Linear added support for emojis to projects. This isn't yet supported and an empty icon shows up in the list. Raycast render standard emojis, but Linear also has custom one from Slack, which should also be renderable (just images at the end). 

## Steps To Reproduce

1. Create a project and assign an emoji
2. Open the Search Projects command
3. See the emoji not rendered

## The current behavior

Emoji isn't rendered

## The expected behavior

Emoji is rendered (custom and standard ones)

### Raycast version

Version:

### PR DESCRIPTION
**Pull request by thomaslombart**
## Description

Add support for emojis in Linear projects.

Closes #2695.

## Screencast

CleanShot 2022-09-20 at 11 00 17@2x

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/linear/src/helpers/projects.ts [hunk 1]
```diff
@@ -40,3 +43,20 @@ export const projectStatusText: Record<string, string> = {
...
+
+export function getProjectIcon(project?: Pick<Project, "icon" | "color">) {
+  if (!project) {
+    return { source: { light: "light/no-project.svg", dark: "dark/no-project.svg" } };
+  }
+
+  const emojiRegex = new RegExp(/:[a-z0-9_+-]+:/, "g");
+
+  if (project.icon && emojiRegex.test(project.icon)) {
+    return emojis.get(project.icon);
```
Comment (mathieudutour) [inline extensions/linear/src/helpers/projects.ts:55]: there are no colors when you have an emoji?

Comment (thomaslombart) [reply extensions/linear/src/helpers/projects.ts:55]: No need to set a color as Raycast as the source of the image can be a single emoji (cf [Image.Source](https://developers.raycast.com/api-reference/user-interface/icons-and-images#image.source)) 🙂

Comment (mathieudutour) [reply extensions/linear/src/helpers/projects.ts:55]: yeah I know, I was wondering if you could specify a color with an emoji on Linear. Turns out no but I was too lazy to check 😄

**Review by mathieudutour — APPROVED** [review]

File: extensions/linear/src/helpers/projects.ts (hunk 1 above)
Comment (thomaslombart) [reply extensions/linear/src/helpers/projects.ts:55]: Ah, my bad 😄

**Pull request merged**